core-java/src/test/java/org/baeldung/java/sorting/JavaSorting.java

```diff
@@ -0,0 +1,229 @@
+package org.baeldung.java.sorting;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.junit.Test;
+
+public class JavaSorting {
+    
+    @Test
+    public void givenIntArray_whenUsingSort_thenSortedArray() {
+        int [] numbers = new int[] { 5, 1, 89, 255, 7, 88, 200, 123, 66 }, 
+          sortedInts = {1, 5, 7, 66, 88, 89, 123, 200, 255};
+        
+        Arrays.sort(numbers);
+
+        assertTrue(Arrays.equals(numbers, sortedInts));
+    }
+
+    @Test
+    public void givenArrayObjects_whenUsingSort_thenSortedArrayObjects() {
+        Integer[] integers = new Integer[] 
+          { 5, 1, 89, 255, 7, 88, 200, 123, 66 }, 
+            sortedIntegers = {1, 5, 7, 66, 88, 89, 123, 200, 255};
+        
+        
+        Arrays.sort(integers, new Comparator<Integer>() {
+            @Override
+            public int compare(Integer a, Integer b) {
+                return a - b;
+            }
+        });
+        
+        assertTrue(Arrays.equals(integers, sortedIntegers));
+    }
+
+    @Test
+    public void givenArray_whenUsingSortWithLambdas_thenSortedArray() {
+        Integer[] integers = new Integer[] 
+            { 5, 1, 89, 255, 7, 88, 200, 123, 66 }, 
+              sortedIntegers = {1, 5, 7, 66, 88, 89, 123, 200, 255};
+        
+        Arrays.sort(integers, (a, b) -> {
+            return a - b;
+        });
+        
+        assertTrue(Arrays.equals(integers, sortedIntegers));
+    }
+
+    @Test
+    public void givenEmpArray_SortEmpArray_thenSortedArrayinNaturalOrder() {
+        Employee[] employees = new Employee[] { 
+          new Employee("John", 23, 5000), 
+          new Employee("Steve", 26, 6000), 
+          new Employee("Frank", 33, 7000),
+          new Employee("Earl", 43, 10000), 
+          new Employee("Jessica", 23, 4000), 
+          new Employee("Pearl", 33, 6000)};
+        String sortedArrayString = "[(John,23,5000.0), (Jessica,23,4000.0), "
+          + "(Steve,26,6000.0), (Frank,33,7000.0), (Pearl,33,6000.0), "
+          + "(Earl,43,10000.0)]";
+        
```

Comment:
I thought of that. But it would increase lines of code. Since individual arrays needs to be created. @pkhanal